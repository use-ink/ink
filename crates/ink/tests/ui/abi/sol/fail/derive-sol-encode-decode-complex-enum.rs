#![allow(unexpected_cfgs)]

// `SolEncode` and `SolDecode` derive macros don't generate implementations for
// enums with fields (see respective docs for rationale).

#[derive(ink::SolDecode, ink::SolEncode)]
enum EnumWithUnnamedFields {
    Unit,
    UnnamedFields(bool, u8, String),
}

#[derive(ink::SolDecode, ink::SolEncode)]
enum EnumWithNamedFields {
    Unit,
    NamedFields {
        status: bool,
        count: u8,
        reason: String,
    },
}

fn main() {}
