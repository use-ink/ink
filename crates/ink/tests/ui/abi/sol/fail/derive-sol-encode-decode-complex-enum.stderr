error: can only derive `SolDecode` for Rust `enum` items with only unit-only or field-less variants
  --> tests/ui/abi/sol/fail/derive-sol-encode-decode-complex-enum.rs:7:1
   |
7  | / enum EnumWithUnnamedFields {
8  | |     Unit,
9  | |     UnnamedFields(bool, u8, String),
10 | | }
   | |_^

error: can only derive `SolEncode` for Rust `enum` items with only unit-only or field-less variants
  --> tests/ui/abi/sol/fail/derive-sol-encode-decode-complex-enum.rs:7:1
   |
7  | / enum EnumWithUnnamedFields {
8  | |     Unit,
9  | |     UnnamedFields(bool, u8, String),
10 | | }
   | |_^

error: can only derive `SolDecode` for Rust `enum` items with only unit-only or field-less variants
  --> tests/ui/abi/sol/fail/derive-sol-encode-decode-complex-enum.rs:13:1
   |
13 | / enum EnumWithNamedFields {
14 | |     Unit,
15 | |     NamedFields {
16 | |         status: bool,
...  |
19 | |     },
20 | | }
   | |_^

error: can only derive `SolEncode` for Rust `enum` items with only unit-only or field-less variants
  --> tests/ui/abi/sol/fail/derive-sol-encode-decode-complex-enum.rs:13:1
   |
13 | / enum EnumWithNamedFields {
14 | |     Unit,
15 | |     NamedFields {
16 | |         status: bool,
...  |
19 | |     },
20 | | }
   | |_^
